frontend/src/pages/destructionlist/detail/DestructionListDetail.tsx

```diff
@@ -82,16 +89,63 @@ function getDisplayableList(
  * Destruction list detail page
  */
 export function DestructionListDetailPage() {
-  const { destructionList, reviewers } =
+  const { destructionList, reviewers, archivists } =
     useLoaderData() as DestructionListDetailContext;
 
+  const [modalOpenState, setModalOpenState] = useState(false);
+
+  const modalFormFields: FormField[] = [
+    {
+      label: "Arhivaris",
+      name: "assigneeIds",
+      options: archivists.map((user) => ({
+        value: String(user.pk),
+        label: user.username,
+      })),
+      required: true,
+    },
+  ];
+
   // TODO - Make a 404 page
   if (!destructionList) return <div>Deze vernietigingslijst bestaat niet.</div>;
 
+  const onSubmit = async (_: FormEvent, data: SerializedFormData) => {
+    await markDestructionListAsFinal(destructionList.uuid, {
+      user: Number(data.assigneeIds),
+    });
+    setModalOpenState(false);
+    return redirect("/");
+  };
+
   return (
     <CardBaseTemplate>
       <Body>
-        <H2>{destructionList.name}</H2>
+        <Grid>
+          <Column span={2}>
+            <H1>{destructionList.name}</H1>
+          </Column>
+          {destructionList.status === "internally_reviewed" && (
+            <Column span={8}>
+              <Modal
+                title="Mark as final"
+                open={modalOpenState}
+                size="m"
+                onClose={() => setModalOpenState(false)}
+              >
+                <Body>
+                  <Form
+                    fields={modalFormFields}
+                    onSubmit={onSubmit}
+                    validateOnChange={true}
+                  />
+                </Body>
+              </Modal>
+              <Button type="submit" onClick={() => setModalOpenState(true)}>
```

Comment:
Can we maybe move use the "secondary navigation" for the button? See: https://www.chromatic.com/component?appId=659d1ab1b20571e0dbb5e47c&csfId=templates-list--with-secondary-navigation&buildNumber=379&k=66869a296fb6e686266ed403-dark%20desktop-interactive-true&h=7&b=-2. A nice compact button on the top right.